staging/src/k8s.io/apiextensions-apiserver/third_party/forked/celopenapi/model/schemas_test.go

```diff
@@ -238,3 +238,156 @@ func testSchema() *schema.Structural {
 	}
 	return ts
 }
+
+func TestEstimateMaxLengthJSON(t *testing.T) {
+	type maxLengthTest struct {
+		Name              string
+		InputSchema       *schema.Structural
+		ExpectedMaxLength int64
+	}
+	tests := []maxLengthTest{
+		{
+			Name: "numberArray",
+			InputSchema: &schema.Structural{
+				Generic: schema.Generic{
+					Type: "array",
+				},
+				Items: &schema.Structural{
+					Generic: schema.Generic{
+						Type: "integer",
+					},
+					ValueValidation: &schema.ValueValidation{
+						Format: "int64",
+					},
+				},
+			},
+			// expected JSON is [0,0,...] so our length should be (3000000 - 2) / 2
```

Comment:
It's also okay to just put the calculation as the literal, e.g. `ExpectedMaxLength: (3000000 - 2) / 2)` (but keep the JSON form in the comment)